itest/lnd_sweep_test.go

```diff
@@ -2352,3 +2352,177 @@ func testFeeReplacement(ht *lntest.HarnessTest) {
 	// Finally, clean the mempool.
 	ht.MineBlocksAndAssertNumTxes(1, 1)
 }
+
+// testBumpFeeLowBudget checks that when the requested ideal budget cannot be
+// met, the sweeper still sweeps the input with the actual budget.
+func testBumpFeeLowBudget(ht *lntest.HarnessTest) {
+	// Create a new node with a large `maxfeerate` so it's easier to run the
+	// test.
+	alice := ht.NewNode("Alice", []string{
+		"--sweeper.maxfeerate=10000",
```

Comment:
This maxFeeRate raised the following question: Let's imagine a max feerate of 100 sats/vbyte, we should probably before allocating wallet inputs make sure that we need them in the first place, meaning that the max_fee_rate can maybe already be paid without using the full budget. Wdyt. Whats the point of locking a 100k sats wallet input if the max_fee_rate can already be Borrowed by other sweep inputs ?

Reply:
if it can be borrowed the wallet inputs won't be added.